opensrp-anc/src/main/java/org/smartregister/anc/library/helper/AncRulesEngineFactory.java

```diff
@@ -26,6 +31,39 @@ public AncRulesEngineFactory(Context context, Map<String, String> globalValues,
     protected Facts initializeFacts(Facts facts) {
         super.initializeFacts(facts);
         facts.put("helper", ancRulesEngineHelper);
+        return extractValues(facts);
+    }
+
+    /*
+     *  Extract value from JSON
+     */
+    protected Facts extractValues(Facts facts) {
+        try {
+            for (String key : facts.asMap().keySet()) {
+                Object value = facts.get(key);
+                if (value instanceof String) {
+                    if (((String) value).startsWith("{") && ((String) value).endsWith("}")) {
+                        facts.put(key, new JSONObject(((String) value)).getString("value"));
+                    } else {
+                        facts.put(key, value);
+                    }
+                } else if (value instanceof List) {
+                    JSONArray valuesArray = new JSONArray(value.toString());
+                    ArrayList<String> updatedArray = new ArrayList<>();
+                    for (int i = 0; i < valuesArray.length(); i++) {
+                        String keyVal = valuesArray.optString(i);
+                        if (keyVal.startsWith("{") && keyVal.endsWith("}")) {
+                            updatedArray.add(new JSONObject(keyVal).getString("value"));
```

Comment:
```suggestion
                       JSONObject ourObject=  new JSONObject(keyVal).getString(JsonFormConstance.VALUE);
``` You can check the correct Constants